lib/json/json.c

```diff
@@ -495,7 +493,7 @@ static ptrdiff_t get_elem_size(const struct json_obj_descr *descr)
 		for (i = 0; i < descr->object.sub_descr_len; i++) {
 			ptrdiff_t s = get_elem_size(&descr->object.sub_descr[i]);
 
-			total += ROUND_UP(s, descr->alignment);
+			total += ROUND_UP(s, descr->alignment + 1);
```

Comment:
I am guessing that the +1 here is related to -1 above. Could you add some comments why we are doing this as it is not really obvious (at least for me)..

Reply:
Alignment will never be 0, but will never be more than 4. By subtracting 1, I can make it range from 0-3 and only use 2 bits.

Reply:
Thx, makes sense now.

Reply:
This change converts the alignment into a mask. Could we indicate this in the name of the filed by renaming alignment into alignment_mask (or something else which indicate the difference) ?